src/mudlet.pro

```diff
@@ -119,39 +119,25 @@ BUILD_COMMIT_TEST = $$lower($$BUILD_COMMIT_TEST)
 }
 
 
-########################## Version and Build setting ###########################
-# Set the current Mudlet Version, unfortunately the Qt documentation suggests
-# that only a #.#.# form without any other alphanumberic suffixes is required:
+# Set Mudlet version (update in CMakeLists.txt as well)
 VERSION = 4.17.2
 
-# if you are distributing modified code, it would be useful if you
-# put something distinguishing into the MUDLET_VERSION_BUILD environment
-# variable (it should use '-' as the first character) to make identification of
-# the used version simpler
-# Note: the qmake BUILD variable is NOT a built-in one
-# BUILD = $$(MUDLET_VERSION_BUILD)
-isEmpty( BUILD ) {
-# Possible values are:
-# "-dev" for the development build
-# "-ptb" for the public test build
-# "" for the release build
-# A core dev team member setting things up for a release should comment out the
-# following line - as the app-build.txt file must not contain anything (other
-# than whitespace) for a RELEASE build:
-   # BUILD = "-dev-"$${GIT_SHA1}
-   BUILD = ""
-} else {
-   BUILD = $${BUILD}-$${GIT_SHA1}
+# Set BUILD based on MUDLET_VERSION_BUILD or default
+BUILD = $$(MUDLET_VERSION_BUILD)
+!isEmpty(BUILD) {
+    BUILD = ${BUILD}-${GIT_SHA1}
```

Comment:
You've lost the protection against an empty environmental variable triggering a "Release" type build though that has been retained by the CMake build system!
```suggestion
    BUILD = ${BUILD}-${GIT_SHA1}
} else {
    BUILD = "-dev-"$${GIT_SHA1}
}
```